src/Orleans/Orleans.csproj

```diff
@@ -74,8 +74,9 @@
   </PropertyGroup>
   <ItemGroup>
     <Reference Include="Microsoft.CSharp" />
-    <Reference Include="Newtonsoft.Json">
-      <HintPath>$(SolutionDir)packages\Newtonsoft.Json.6.0.8\lib\net45\Newtonsoft.Json.dll</HintPath>
+    <Reference Include="Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
```

Comment:
Good point @jason-bragg 

Using "soft" name references has been the norm since the start of the Orleans project [which BTW was before NuGet too] and for various "bad experience" reasons that we have been talked about and shared as "folk-lore" both internally and externally in the past, but as you point out we have never actually officially written down those guidelines.

PR #910 was one of the most recent conversations on this topic, which i'll copy a snippet of below for reference.

I think you are right though that we do need to start writing down the project-specific code style / workflow points such as this, so they are easily discoverable. 
I thought i recalled that someone started pulling some of that info together at one point in the gh-pages branch, but hunting around and all i can see at the moment is fairly generic pointers to DottNet CoreCLR docs :(  http://dotnet.github.io/orleans/#contributing-to-this-project

I'll start a new PR to try to consolidate some of this type of "developer info" and circulate for comment.
Sounds good?

----------
https://github.com/dotnet/orleans/pull/910

> > I'm not convinced it's worth it to fight the tools, imperfect as they are.

> Agreed, it is a slightly different issue, but you have no idea what a huge mess the Geo interns got themselves into by "just letting the tools do their thing" At one point nothing at all would run because of mismatching different assembly references between various sub-projects & sub-systems. I had to spend a very long weekend hacking all their project files by hand (along the same lines as PR #911) to get things working again! :cry:

> >because none of us here have experienced it.

> Yes, we never really experienced this on Orleans project, but mostly because I got you all to do assembly references (and project cross references) right in project files from the get-go ;)